hax/treemakers/CorrectedDoubleS1Scatter.py

```diff
@@ -0,0 +1,386 @@
+        
+        ######### Select Interactions for DoubleScatter Event ########
+        
+        # assume one scatter is interactions[0]
+        int_0 = 0
+        s1_0 = interactions[int_0].s1
+        s2_0 = interactions[int_0].s2
+        
+        # find another scatter
+        otherInts = [0,0]  
+        for i, interaction in enumerate(interactions):
+            if (interaction.s1 != s1_0 and interaction.s2 == s2_0 
+               and otherInts[0] == 0):
+                otherInts[0] = i
+            elif (interaction.s1 != s1_0 and interaction.s2 != s2_0 
+                 and otherInts[1] == 0):
+                otherInts[1] = i
+    
+        # Distinction b/w single and double s2 scatters
+        # Cut events without second s1
+        if otherInts[1] != 0:
+            s1_1 = interactions[otherInts[1]].s1
+            s2_1 = interactions[otherInts[1]].s2
+            int_1 = otherInts[1]
+            ds_second_s2 = 1
+        elif otherInts[0] != 0:
+            s1_1 = interactions[otherInts[0]].s1
+            s2_1 = interactions[otherInts[0]].s2
+            int_1 = otherInts[0]
+            ds_second_s2 = 0
+        else: return dict()
+                                
+        # order s1s/interactions by time
+        if peaks[s1_0].center_time <= peaks[s1_1].center_time:
+            s1_a = s1_0
+            s1_b = s1_1
+            s2_a = s2_0
+            s2_b = s2_1
+            int_a = int_0
+            int_b = int_1
+        else:
+            s1_a = s1_1
+            s1_b = s1_0
+            s2_a = s2_1
+            s2_b = s2_0
+            int_a = int_1
+            int_b = int_0
+            
+        # Additional s1s and s2s removed! see v0.1.0
+        
+        
+        result['s1_a']=peaks[s1_a].area
+        result['s1_a_center_time'] = peaks[s1_a].center_time
+        
+        result['s2_a']=peaks[s2_a].area
+        result['s2_a_center_time'] = peaks[s2_a].center_time
+        
+        
+        result['s1_b']=peaks[s1_b].area
+        result['s1_b_center_time'] = peaks[s1_b].center_time
+        
+        result['s2_b']=peaks[s2_b].area
+        result['s2_b_center_time'] = peaks[s2_b].center_time
+        
+        
+        # Drift Time 
+        
+        result['int_a_drift_time']=interactions[int_a].drift_time
+        result['int_b_drift_time']=interactions[int_b].drift_time
+
+        # Pax position (TpF)
+       
+        result['int_a_x_pax']=interactions[int_a].x
+        result['int_a_y_pax']=interactions[int_a].y
+        result['int_a_z_pax']=interactions[int_a].z
+
+        result['int_b_x_pax']=interactions[int_b].x
+        result['int_b_y_pax']=interactions[int_b].y
+        result['int_b_z_pax']=interactions[int_b].z
+
+######### Compute DoubleScatter Specific Variables #########
+            
+        # Select largest hits on each channel in s10 and s11 peaks
+        s1_a_hitChannels = []
+        s1_a_hitAreas = []
+        
+        for hit in peaks[s1_a].hits:
```

Comment:
Isn't this a bit verbose? The thing you want in the end is the number of distinct channels. There are ways to get there quicker i.e. https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.setdiff1d.html

Reply:
There's certainly more straightforward ways, this just required the least thought when I originally put this together. Maybe better to move forward with adding corrections for now and consider a more thorough re-write soon.